packages/block-library/src/search/edit.native.js

```diff
@@ -204,24 +235,68 @@ export default function SearchEdit( {
 		isLongButton && { flexDirection: 'column' },
 	];
 
-	const getPlaceholderAccessibilityLabel = () => {
+	/**
+	 * If a screenreader is enabled, create a descriptive label for this field. If
+	 * not, return a label that is used during automated UI tests.
+	 *
+	 * @return {string} The accessibilityLabel for the Search Button
+	 */
+	const getAccessibilityLabelForButton = () => {
+		if ( ! isScreenReaderEnabled ) {
+			return 'search-block-button';
+		}
+
+		return `${ __(
+			'Search button. Current button text is'
+		) } ${ buttonText }`;
+	};
+
+	/**
+	 * If a screenreader is enabled, create a descriptive label for this field. If
+	 * not, return a label that is used during automated UI tests.
+	 *
+	 * @return {string} The accessibilityLabel for the Search Input
+	 * 					 placeholder field.
+	 */
+	const getAccessibilityLabelForPlaceholder = () => {
+		if ( ! isScreenReaderEnabled ) {
+			return 'search-block-input';
+		}
+
 		const title = __( 'Search input field.' );
 		const description = placeholder
 			? `${ __( 'Current placeholder text is' ) } ${ placeholder }`
 			: __( 'No custom placeholder set' );
 		return `${ title } ${ description }`;
 	};
 
+	/**
+	 * If a screenreader is enabled, create a descriptive label for this field. If
+	 * not, return a label that is used during automated UI tests.
+	 *
+	 * @return {string} The accessibilityLabel for the Search Label field
+	 */
+	const getAccessibilityLabelForLabel = () => {
+		if ( ! isScreenReaderEnabled ) {
+			return 'search-block-label';
+		}
+
+		return `${ __( 'Search block label. Current text is' ) } ${ label }`;
+	};
+
 	const renderTextField = () => {
 		return (
 			<View
 				style={ styles.searchInputContainer }
 				accessible={ true }
 				accessibilityRole="none"
-				accessibilityHint={ __(
-					'Double tap to edit placeholder text'
-				) }
-				accessibilityLabel={ getPlaceholderAccessibilityLabel() }
+				accessibilityHint={
+					isScreenReaderEnabled
+						? __( 'Double tap to edit placeholder text' )
+						: null
+				}
+				accessibilityLabel={ getAccessibilityLabelForPlaceholder() }
+				testID="search-block-input"
```

Comment:
Is it accurate to say the `testID`s in this file are unused until `react-native` > 0.65 plus refactoring or am I misunderstanding?

Reply:
That's a good point. I've removed `testID` now that it's not being used in 17a78d9fcfd60f5716dc00034e1ae42a373dc394. We can add it back when we upgrade RN.